website/components/NavSection/index.js

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react';
+import Link from 'next/link';
+import PropTypes from 'prop-types';
+import { useRouter } from 'next/router';
+import { SubNavSection, SubNavLink } from '@strapi/design-system/SubNav';
+
+const NavSection = ({ title, pages }) => {
+  const [isSectionActive, setIsSectionActive] = useState(false);
+  const router = useRouter();
+  useEffect(() => {
+    const pageLinks = pages.map((page) => page.link);
+    setIsSectionActive(pageLinks.includes(`/${router.query.slug}`));
+  }, [router]);
```

Comment:
I don't think you need an intermediate state modification. WYT of deriving the value instead of running in an effect?

```suggestion
    const pageLinks = pages.map((page) => page.link);
    const isSectionActive = pageLinks.includes(`/${router.query.slug}`);
```

Reply:
(or even using a `.find` instead)